modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/components/LayoutReports.js

```diff
@@ -13,82 +13,117 @@
  */
 
 import ClayAlert from '@clayui/alert';
-import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayButton from '@clayui/button';
+import ClayProgressBar from '@clayui/progress-bar';
 import {useIsMounted} from '@liferay/frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
 import PropTypes from 'prop-types';
-import React, {useCallback, useEffect, useReducer} from 'react';
+import React, {useCallback, useContext, useEffect, useState} from 'react';
 
+import {StoreDispatchContext, StoreStateContext} from '../context/StoreContext';
+import APIService from '../utils/APIService';
 import BasicInformation from './BasicInformation';
 import EmptyLayoutReports from './EmptyLayoutReports';
-
-const initialState = {
-	data: null,
-	error: null,
-	loading: false,
-};
-
-const dataReducer = (state, action) => {
-	switch (action.type) {
-		case 'LOAD_DATA':
-			return {
-				...state,
-				loading: true,
-			};
-
-		case 'SET_ERROR':
-			return {
-				...state,
-				error: action.error,
-				loading: false,
-			};
-
-		case 'SET_DATA':
-			return {
-				data: {
-					...action.data,
-				},
-				error: action.data?.error,
-				loading: false,
-			};
-
-		default:
-			return initialState;
-	}
-};
+import LayoutReportsIssuesList from './LayoutReportsIssuesList';
 
 export default function LayoutReports({
 	eventTriggered,
 	isPanelStateOpen,
 	layoutReportsDataURL,
+	portletNamespace,
 }) {
 	const isMounted = useIsMounted();
 
-	const [state, dispatch] = useReducer(dataReducer, initialState);
+	const {data, error, loading} = useContext(StoreStateContext);
+
+	const dispatch = useContext(StoreDispatchContext);
 
 	const safeDispatch = useCallback(
 		(action) => {
 			if (isMounted()) {
 				dispatch(action);
 			}
 		},
-		[isMounted]
+		[dispatch, isMounted]
 	);
 
+	const [percentage, setPercentage] = useState(0);
+
+	useEffect(() => {
+		if (loading && !error) {
+			const x = Date.now();
```

Comment:
![come-on](https://user-images.githubusercontent.com/7074/116525614-fdd61480-a8d8-11eb-821e-78ee92385201.gif)

This should be `initial` or something descriptive, not `x`.

But seeing as you have a green CI run, I'd just loop back and do it next time you touch this file, if you do.

Reply:
Thanks! I'll change it and push force 👍